src/OmnichannelChatSDK.ts

```diff
@@ -528,46 +554,95 @@ class OmnichannelChatSDK {
 
         try {
             await this.conversation!.sendFileMessage(fileMetadata, messageToSend);
+            this.scenarioMarker.completeScenario(TelemetryEvent.UploadFileAttachment);
             return messageToSend;
         } catch (error) {
             console.error(`OmnichannelChatSDK/uploadFileAttachment/error: ${error}`);
+            this.scenarioMarker.failScenario(TelemetryEvent.UploadFileAttachment);
             return error;
         }
     }
 
     public async downloadFileAttachment(fileMetadata: IFileMetadata): Promise<Blob> {
-        return this.conversation!.downloadFile(fileMetadata);
+        this.scenarioMarker.startScenario(TelemetryEvent.DownloadFileAttachment, {
+            RequestId: this.requestId,
+            ChatId: this.chatToken.chatId as string
+        })
+
+        try { 
+            const downloadedFile = this.conversation!.downloadFile(fileMetadata);
+            this.scenarioMarker.completeScenario(TelemetryEvent.DownloadFileAttachment);
+            return downloadedFile;
+        }
+        catch(e) {
+            this.scenarioMarker.failScenario(TelemetryEvent.DownloadFileAttachment);
+            return this.conversation!.downloadFile(fileMetadata);
```

Comment:
We should return `new Error('DownloadFileAttachmentFailed')` on failure

Reply:
Syntax did not work, but I found something online that does the same thing